### PR TITLE
Adding ion did method

### PR DESCRIPTION
**Pull request by nitro-neal**
Example ion did doc:

```
    {
      didDocument: {
        id: 'did:ion:EiBSi4TaOJHdopyIENKy6M54enbJlFJjhLzkKCzglId1AQ',
        recoveryPrivateJWK: '{EC secp256k1 Re9aN3E6roeH29V3XdTox9gkaGLxRGLlrARVhbQDdQU ELHR7j8Bo-yjZbrKZGzsTK7PGdhhuhHp9cfJ0-UmadQ     ES256K 2e01f740-c1a2-499d-8daa-8553d0f79486 EvsJOQYYx23_XBSoJ3ILuQXYupxX6175D9ZFKqPkUyw     }',
        longForm: 'did:ion:EiBSi4TaOJHdopyIENKy6M54enbJlFJjhLzkKCzglId1AQ:eyJkZWx0YSI6eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljS2V5cyI6W3siaWQiOiJmZTE5ZTQ2OS04ZGQ1LTQ1YjItOWYwYi1mNTgzYmQzNTgxNDEiLCJwdWJsaWNLZXlKd2siOnsiYWxnIjoiRWREU0EiLCJjcnYiOiJFZDI1NTE5Iiwia2lkIjoiOWZhYzZhYzMtNjBjYy00MTI5LWJmNWUtMTEwNTNiMDcwYzNmIiwia3R5IjoiT0tQIiwieCI6IkhoT3hKanJ0enVabVl6WlE0NzN0MG4zRFRYcGpsZjJpcXhWVEVUMllCb28ifSwicHVycG9zZXMiOlsiYXV0aGVudGljYXRpb24iLCJhc3NlcnRpb25NZXRob2QiXSwidHlwZSI6IkVkMjU1MTkifV19fV0sInVwZGF0ZUNvbW1pdG1lbnQiOiJFaUJNX1lRWjN1R2Q0UWFqQjRBTTJDN29aeUxzY0REM3FDck1KSTUxZUJiQ1NRIn0sInN1ZmZpeERhdGEiOnsiZGVsdGFIYXNoIjoiRWlCVS1ubVVzWnRoU24zMTJ1T2VUMDgyR3RyNmQ5WFRMTjViWVJzTTVoSGJhUSIsInJlY292ZXJ5Q29tbWl0bWVudCI6IkVpRHprM3FRdVZWaHZIT0g1S05yY1U1eVFrWkhzekJ2STJGcV9jVkpqWm8xTncifX0'
      },
      privKeyJWK: '{"kty":"OKP","crv":"Ed25519","x":"HhOxJjrtzuZmYzZQ473t0n3DTXpjlf2iqxVTET2YBoo","alg":"EdDSA","kid":"9fac6ac3-60cc-4129-bf5e-11053b070c3f","d":"5NUx6WoojFyOKM8XjdyzCAa9VyIrZpKly4foVao2tt8"}'
    }
```

This change also:
* updated the ssi-sdk to its lastest published version
* updated all methods to use new ssi-sdk version
* more unit tests and interface changes
* jwk is now the private key instead of base58 priv key

image

### CODE REVIEW

File: wasm/main.go [hunk 1]
```diff
@@ -3,101 +3,194 @@
...
 		resultObj, err := simplifyForJS(result)
 		if err != nil {
 			return generateError(err)
 		}
 
-		// Create a JavaScript object and set its properties using the values from the Go struct
 		jsDIDObj := js.Global().Get("Object").New()
 		jsDIDObj.Set("didDocument", js.ValueOf(resultObj))
-		jsDIDObj.Set("privateKeyBase58", js.ValueOf(string(base58PrivKey)))
+		jsDIDObj.Set("privKeyJWK", js.ValueOf(privKeyJWKString))
```
Comment (thereisnogabe) [inline wasm/main.go:73]: should this be `privKeyJwk`?

File: wasm/main.go [hunk 2]
```diff
@@ -3,101 +3,194 @@
...
-// @Summary Resolve a DID
-// @Description Resolve a given DID using a set of resolvers (KeyResolver, WebResolver, PKHResolver, and PeerResolver), and return the corresponding DID document as a JavaScript object
-// @Param didString string "The DID string to be resolved"
-// @Success js.Object "{ <DID document fields> }"
+// @Summary     Create DID:Ion pair
+// @Description Generate a DID Ion pair (using Ed25519) for ION protocol and return a JavaScript object containing the ION document and the private JWK
+// @Success     js.Object "{ didDocument: <object>, privKeyJWK: <string> }"
 // @Error js.Value "An error object with a message describing the error"
-func resolveDID() js.Func {
+func createDIDIon() js.Func {
```
Comment (thereisnogabe) [inline wasm/main.go]:
```suggestion
func createDIDION() js.Func {
```

File: wasm/main.go [hunk 3]
```diff
@@ -3,101 +3,194 @@
...
+func resolveDID() js.Func {
+	return js.FuncOf(func(this js.Value, topArgs []js.Value) interface{} {
+		promiseConstructor := js.Global().Get("Promise")
+		return promiseConstructor.New(js.FuncOf(func(_ js.Value, args []js.Value) interface{} {
+			resolve := args[0]
+			reject := args[1]
+
+			go func() {
+				if len(topArgs) != 1 {
+					reject.Invoke(generateError(errors.New("invalid arg count, need did as argument")))
```
Comment (thereisnogabe) [inline wasm/main.go:149]: could you break this out into two lines?

File: wasm/main.go [hunk 4]
```diff
@@ -3,101 +3,194 @@
...
+				}
+
+				resultBytes, err := json.Marshal(doc)
+				if err != nil {
+					reject.Invoke(generateError(err))
+					return
+				}
+
+				var resultObj map[string]any
+				err = json.Unmarshal(resultBytes, &resultObj)
```
Comment (thereisnogabe) [inline wasm/main.go:183]: can combine these

**Review by thereisnogabe — APPROVED** [review]

File: wasm/main.go (hunk 1 above)
Comment (nitro-neal) [reply wasm/main.go:73]: good catch

**Pull request merged**